TestStandardSDKIntegrationTestSuite flake

Flake: https://github.com/celestiaorg/celestia-app/actions/runs/25882167492/job/76064670385#step:4:23

```
--- FAIL: TestStandardSDKIntegrationTestSuite (15.82s)
    std_sdk_test.go:64: setting up integration test suite
    --- FAIL: TestStandardSDKIntegrationTestSuite/TestStandardSDK (12.53s)
        --- FAIL: TestStandardSDKIntegrationTestSuite/TestStandardSDK/multiple_send_sdk.Msgs_in_one_sdk.Tx (1.01s)
            std_sdk_test.go:362: 
                	Error Trace:	/home/runner/work/celestia-app/celestia-app/app/test/std_sdk_test.go:362
                	Error:      	Received unexpected error:
                	            	rpc error: code = NotFound desc = tx not found: 53BCE023CD922D2BEBEE61BEF99F053397C5B6D6D8EF51093A2B878F7D465879
                	Test:       	TestStandardSDKIntegrationTestSuite/TestStandardSDK/multiple_send_sdk.Msgs_in_one_sdk.Tx
    network.go:74: tearing down testnode
    network.go:115: error when closing API server close tcp 127.0.0.1:57978: use of closed network connection
FAIL
FAIL	github.com/celestiaorg/celestia-app/v9/app/test	18.561s
FAIL
Error: Process completed with exit code 1.
```

--- a/app/test/std_sdk_test.go
+++ b/app/test/std_sdk_test.go
@@ -1,6 +1,7 @@
 package app_test
 
 import (
+	"context"
 	"sync"
 	"testing"
 	"time"
@@ -349,21 +350,30 @@ func (s *StandardSDKIntegrationTestSuite) TestStandardSDK() {
 
 				switch txError := err.(type) {
 				case *user.ExecutionError:
-					txResp, err := serviceClient.GetTx(s.cctx.GoContext(), &sdktx.GetTxRequest{Hash: txError.TxHash})
-					require.NoError(t, err)
-					assert.Equal(t, tt.expectedCode, txError.Code, txResp.TxResponse.RawLog)
+					assert.Equal(t, tt.expectedCode, txError.Code, fetchRawLog(s.cctx.GoContext(), serviceClient, txError.TxHash))
 				case *user.BroadcastTxError:
 					assert.Equal(t, tt.expectedCode, txError.Code, txError.ErrorLog)
 				}
 			} else {
 				require.NoError(t, err)
 				require.NotNil(t, res)
-				getTxResp, err := serviceClient.GetTx(s.cctx.GoContext(), &sdktx.GetTxRequest{Hash: res.TxHash})
-				require.NoError(t, err)
-				assert.Equal(t, tt.expectedCode, res.Code, getTxResp.TxResponse.RawLog)
+				assert.Equal(t, tt.expectedCode, res.Code, fetchRawLog(s.cctx.GoContext(), serviceClient, res.TxHash))
 			}
 		})
 	}
+}
+
+// fetchRawLog returns the raw log for txHash on a best-effort basis. Used only
+// to enrich assert failure messages; it must not fail the test if the cosmos
+// tx indexer hasn't caught up yet (celestia's TxStatus reports COMMITTED via
+// BlockStore.SaveTxInfo before the async IndexerService populates the index
+// GetTx reads, so calls right after SubmitTx can race and return NotFound).
+func fetchRawLog(ctx context.Context, serviceClient sdktx.ServiceClient, txHash string) string {
+	resp, err := serviceClient.GetTx(ctx, &sdktx.GetTxRequest{Hash: txHash})
+	if err != nil || resp == nil || resp.TxResponse == nil {
+		return ""
+	}
+	return resp.TxResponse.RawLog
 }
 
 func (s *StandardSDKIntegrationTestSuite) TestGRPCQueries() {

